### PR TITLE
Add factory method for remote task with fixed version

### PR DESCRIPTION
**Pull request by sonjaer**
# TL;DR
Factory method to create remote task plan with fixed version

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added - NA there are already tests for it
 - [ ] Code documentation added NA
 - [ ] Any pending items have an associated Issue NA

### CODE REVIEW

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java [hunk 1]
```diff
@@ -76,15 +78,35 @@ public static <InputT, OutputT> SdkRemoteTask<InputT, OutputT> create(
...
+
+  /**
+   * Create a remote task, a reference to a task that have been deployed previously.
+   *
+   * @param domain the domain of the remote launch plan
+   * @param project the project of the remote launch plan
+   * @param name the name of the remote launch plan
+   * @param inputs the {@link SdkType} for the inputs of the remote launch plan
+   * @param outputs the {@link SdkType} for the outputs of the remote launch plan
+   * @return the remote task
```
Comment (sonjaer) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java:101]:
```suggestion
   * @return the latest version of the remote task
```

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java [hunk 2]
```diff
@@ -76,6 +77,7 @@ public static <InputT, OutputT> SdkRemoteTask<InputT, OutputT> create(
       String domain,
       String project,
       String name,
+      @Nullable String version,
```
Comment (honnix) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java:80]: This is not set into SdkRemoteTask in the method body.

Comment (sonjaer) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java:80]: :face-plant:

Comment (sonjaer) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteTask.java:80]: fixed

**Review by honnix — APPROVED** [review]

**Pull request merged**